reVX/utilities/reeds_cols.py

```diff
@@ -179,16 +188,17 @@ def add_reeds_columns(supply_curve_fpath, out_fp=None, capacity_col="capacity",
         By default, ``"capacity"``.
     extra_data : list of dicts
         A list of dictionaries, where each dictionary contains two keys.
-        The first key is "data_fp", and it points to the path where the
-        extra data is being extracted from. This must be an HDF5 or JSON
-        file (i.e. must end in ".h5" or ".json"). The second key is
-        "dsets", and it points to a list of dataset names to extract
-        from the file. For JSON data extraction, the values of the
-        datasets must either be scalars or must match the length of the
-        input supply curve. For HDF5 data, the datasets must be 1D
-        datasets, and they will be merged with the input supply curve
-        on ``merge_col`` (column must be in the HDF5 file meta).
-        By default, ``None``.
+        The first key is "data_fp", and it points either a dictionary of
+        new field/new value pairs or to the path where the extra data is being
+        extracted from. This must be a dictionary, an HDF5, or JSON
+        file (i.e., if not a dictionary, it must end in ".h5" or ".json"). The
+        second key is "dsets", and it points to a list of dataset names to
+        extract from data_fp. For JSON and dictionary data extraction, the
+        values of the datasets must either be scalars or must match the length
+        of the input ``data_frame``. For HDF5 data, the datasets must be 1D
+        datasets, and they will be merged with the input ``data_frame``
+        on ``merge_col`` (column must be in the HDF5 file meta). By default,
+        ``None``.
```

Comment:
Can you add documentation for the `regions` input parameter at the end of this docstring? It doesn't have to be anything long or fancy, but this is the text that gets put into the documentation for this function, so it's important that we have all of our parameters documents.